src/Bicep.Core/TypeSystem/Providers/ResourceTypesProviderDescriptor.cs

```diff
@@ -1,33 +1,47 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Bicep.Core.Registry;
+using Bicep.Core.Semantics.Namespaces;
+
 namespace Bicep.Core.TypeSystem.Providers;
 
-public record ResourceTypesProviderDescriptor
+public class ResourceTypesProviderDescriptor
 {
+    public const string LegacyVersionPlaceholder = "1.0.0";
+
     public ResourceTypesProviderDescriptor(
-        string name,
-        string version,
-        bool isImplicitImport,
-        string? alias = null,
-        string? artifactRef = null,
-        Uri? typesBaseUri = null)
+        string NamespaceIdentifier,
+        Uri ParentModuleUri,
+        string Version = LegacyVersionPlaceholder,
+        string? Alias = null,
+        ArtifactReference? ArtifactReference = null,
+        Uri? TypesDataUri = null)
     {
-
-        Name = name;
-        Alias = alias ?? name;
-        ArtifactReference = artifactRef ?? (isImplicitImport ? name : $"{name}@{version}");
-        Version = version;
-        TypesBaseUri = typesBaseUri;
+        this.Name = NamespaceIdentifier;
+        this.ParentModuleUri = ParentModuleUri;
+        this.Version = Version;
+        this.IsBuiltIn = IsBuiltInNamespace(NamespaceIdentifier, Version);
+        this.Alias = Alias ?? NamespaceIdentifier;
+        this.ArtifactReference = ArtifactReference;
+        this.TypesDataFileUri = TypesDataUri;
```

Comment:
Elsewhere it's referred to as `typesTgzUri` - what about converging on that as a name?